apex/parallel/optimized_sync_batchnorm_kernel.py

```diff
@@ -0,0 +1,68 @@
+import torch
+from torch.autograd.function import Function
+
+import syncbn
+
+class SyncBatchnormFunction(Function):
+
+    @staticmethod
+    def forward(ctx, input, weight, bias, running_mean, running_variance, eps, track_running_stats = False, momentum = 1.0):
+        torch.cuda.nvtx.range_push("sync_BN_fw")
+        mean, var, var_biased = syncbn.welford_mean_var(input)
```

Comment:
it would be good to have comments on what all these functions do (welford_mean_var, welford_parallel etc), either here or in the backend.

Reply:
Also, I don't see you calling .contiguous on input tensors anywhere, and our optimized kernels depend on input/gradOutput being contiguous.

Reply:
I'll comment it at the syncbn.cpp.
Still intend to clean this up a little bit more